rcl/src/rcl/node.c

```diff
@@ -179,7 +211,47 @@ rcl_node_init(
   }
   // actual domain id
   node->impl->actual_domain_id = domain_id;
-  node->impl->rmw_node_handle = rmw_create_node(name, local_namespace_, domain_id);
+
+  const char * ros_enable_security = NULL;
+  const char * ros_enforce_security = NULL;
+
+  if (rcutils_get_env(sros_enable_security_var_name, &ros_enable_security)) {
```

Comment:
Actually looks like the posix `getenv()` need not be thread-safe either: http://pubs.opengroup.org/onlinepubs/9699919799/functions/getenv.html

So we'd have to use the thread-local storage on all platforms and copy the requested value into the thread local storage. That would at least make it thread-safe with concurrent calls to `rcutils_get_env` though it would still not be thread-safe with setenv, but that's less important I think.